client/state/index.js

```diff
@@ -128,7 +128,8 @@ export function createReduxStore( initialState = {} ) {
 		enhancers.push( sitesSync );
 
 		if ( window.app && window.app.isDebug ) {
-			enhancers.push( consoleDispatcher );
+			// this has to be the first enhancer in the chain for the other enhancers to see it
+			enhancers.unshift( consoleDispatcher );
```

Comment:
One way I was planning on cleaning this up is to stop relying on these imperative semantics and refocus the middleware chain in the declarative way it should be.

For example, this change fixes the issue at hand, but doesn't solve the problem that it's unclear which enhancers come in which order.

What if instead we could see a list in sequence identifying enhancers and the conditions in which they come in?

```js
const always = () => true;
const first = ( [ a, b ] ) => a;
const identity = a => a;
const whenInWindow = () => 'object' === typeof window;

const enhancers = [
	[ consoleDispatcher, whenInWindow ],
	[ devToolsExtension, () => whenInWindow() && window.devToolsExtension ],
	[ applyMiddleware( ...middleware ), always ],
	[ sitesSync, whenInWindow ],
].filter( identity ).map( first );
```

or differently

```js
const inWindow = 'object' === typeof window;

const enhancers = [
	inWindow && consoleDispatcher,
	inWindow && window.devToolsExtension && window.devToolsExtension(),
	applyMiddleware( ...middleware ),
	inWindow && sitesSync,
].filter( identity );
```

different approaches but each having the goal of trying to represent the final sequence of enhancers clearly and also the rules which dictate when they will be there.